components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -339,6 +349,40 @@ class AddonsManagerAdapter(
             }
         }
     }
+
+    /**
+     * Refresh only the portion of the list that contains the [addon].
+     * @property addon The add-on to be update.
+     */
+    fun refresh(addon: Addon){
+        addonsMap[addon.id] = addon
+        submitList(createListWithSections(addonsMap.values.toList()))
+    }
+
+    /**
+     * Refresh only the portion of the list that changes between the currentList and the new [addons].
+     * @property addons A list of add-on to replace the actual list.
+     */
+    fun refresh(addons: List<Addon>){
+        addonsMap = addons.associateBy({ it.id }, { it }).toMutableMap()
```

Comment:
I don't understand why you need the `addonsMap` here. You don't seem to be reading from it in this case?

Reply:
It's for keeping  `addonsMap` in sync with the latest changes for futures calls to` refresh(addon: Addon) `

Reply:
Using a map offers faster lockups for when consumers want to just update one add-on (`refresh(addon: Addon)`), if I use a list every time that we want to update one add-on, we need to call `indexOf` and then alter the add-on.

```kotlin
    fun refresh(addon: Addon){
        val index = addonList.indexOf(addon)
        addonList[index] = addon
        submitList(createListWithSections(addonList))
    } 
```

Reply:
Sure, but that's not what I meant :). Maybe the kdocs aren't correct then. `Refresh only the portion of the list that changes between the currentList and the new [addons].` -> The second method that accepts a list refreshes everything, right? Not just the delta / that's what I was trying to find here?

Reply:
In other words, this more of a `refreshAll` function, right? Because if the you only pass in a subset of the add-ons to refresh and you're saving to a new `this.addonsMap`, it would then only have the ones you refreshed, not the entire "list". That's the part I wasn't sure about how you intended this to work.

Reply:
Maybe I should call it `updateData(addon: Addon)` and `updateData(addons: List<Addon>)`, we have two things:

- **The data model** ->  `addonsMap` contains all the add-ons.
- **The visual representation** -> that is kept in the adapter internals, contains the headers + the add-on view items.

Every time, we call `refresh(addon: Addon)` or  `refresh(addons: List<Addon>)`.
`submitList` is calculating a diff and only applying changes to elements that have changed in the **visual representation**, but we don't know exactly what have changed in the **data model**, at least when calling `refresh(addons: List<Addon>)` for this reason I'm updating `addonsMap` with the data.

Reply:
Yeah, I think that's better! I would basically point that out in the kDocs that the list must contain all add-ons but the UI supports partial updates on only the add-ons that have changed.

Alternative: `updateAddon` and `updateAddons`.